video_xblock/static/js/toggle-button.js

```diff
@@ -4,147 +4,159 @@
  */
 
 domReady(function() {
-
-  var MenuItem = videojs.getComponent('MenuItem');
-  var ToggleMenuItem = videojs.extend(MenuItem, {
-    // base class for create buttons for caption and transcripts
-    constructor: function constructor(player, options) {
-      MenuItem.call(this, player, options);
-      this.on('click', this.onClick);
-      this.createEl();
-    },
-    enabledEventName: function enabledEventName() {
-      return this.options_['enabledEvent'];
-    },
-    disabledEventName: function disabledEventName() {
-      return this.options_['disabledEvent'];
-    },
-    createEl: function createEl(type, props, attributes) {
-      props = props || {};
-      var el = MenuItem.prototype.createEl.call(this, arguments.tag, props, attributes);
-      el.setAttribute('role', 'menuitem');
-      el.setAttribute('aria-live', 'polite');
-      el.setAttribute('data-lang', this.options_.track.language);
-      return el;
-    },
-    onClick: function onClick(event) {
-      var menuItem = this.$$('.vjs-menu-item', this.el_.parentNode);
-      var el = event.currentTarget;
-      Array.from(menuItem).forEach(function(caption) {
-        caption.classList.remove('vjs-selected');
-      });
-      el.classList.add('vjs-selected');
-
-      var tracks = this.player_.textTracks();
-      for (var i = 0; i < tracks.length; i++) {
-        var track = tracks[i];
-        if (track.kind === 'captions') {
-          this.player_.captionsLanguage = el.dataset.lang;
-          track.mode = track.language === this.player_.captionsLanguage ? 'showing': 'disabled';
+    'use strict';
+
+    var MenuItem = videojs.getComponent('MenuItem');
+    var ToggleMenuItem = videojs.extend(MenuItem, {
```

Comment:
Increased docstring coverage for all the JS files. As discussed with @z4y4ts, docstrings will be added to all the functions in the next PR (considered as a separate task), as some JS files represent big bunches of functionality, e.g. [ studio_edit.js ](https://github.com/raccoongang/xblock-video/blob/feature/HLSe/video_xblock/static/js/studio-edit.js)